app/src/pages/playground/usePlaygroundPrompt.tsx

```diff
@@ -0,0 +1,211 @@
+    model: {
+      ...newInstance.model,
+      modelName: promptVersion.modelName,
+      provider:
+        openInferenceModelProviderToPhoenixModelProvider(
+          promptVersion.modelProvider
+        ) || DEFAULT_MODEL_PROVIDER,
+    },
+    template: {
+      __type: "chat",
+      messages:
+        "messages" in promptVersion.template
+          ? promptVersion.template.messages.map((m) => ({
+              ...m,
+              id: generateMessageId(),
+              role: getChatRole(m.role?.toLocaleLowerCase() as string),
+            }))
+          : [],
+    },
+    tools: promptVersion.tools.map((t) => ({
+      id: generateToolId(),
+      definition: t.definition,
+    })),
+  } satisfies Partial<PlaygroundInstance>;
+};
+
+export const instanceToPromptVersion = (instance: PlaygroundInstance) => {
+  if (!instance.prompt) {
+    return null;
+  }
+  if (instance.template.__type === "text_completion") {
+    // eslint-disable-next-line no-console
+    console.warn(
+      "Instance to prompt version conversion not supported for text completion"
+    );
+    return null;
+  }
+  return {
+    id: instance.prompt.id,
+    modelName: instance.model.modelName || DEFAULT_MODEL_NAME,
+    modelProvider: instance.model.provider,
+    templateType: "CHAT",
+    template: {
+      __typename: "PromptChatTemplate",
+      messages: instance.template.messages.map((m) => ({
+        content: m.content,
+        role: chatMessageRoleToPromptMessageRole(m.role),
+      })),
+    },
+    tools: instance.tools.map((t) => ({
+      __typename: "ToolDefinition",
+      definition: t.definition,
+    })),
+    // @TODO(apowell): Add description and invocationParameters
+  } satisfies Partial<PromptVersion>;
+};
+
+const getLatestPromptVersion = (
+  prompt: usePlaygroundPromptQuery$data["prompt"]
+) => {
+  if (!prompt) {
+    return null;
+  }
+  return prompt?.promptVersions?.edges?.[0]
+    ?.promptVersion as Mutable<PromptVersion> | null;
+};
+
+const query = graphql`
+  query usePlaygroundPromptQuery($promptId: GlobalID!) {
+    prompt: node(id: $promptId) {
+      ... on Prompt {
+        id
+        name
+        createdAt
+        description
+        promptVersions(first: 1) {
+          edges {
+            promptVersion: node {
+              id
+              description
+              modelName
+              modelProvider
+              invocationParameters
+              templateType
+              template {
+                __typename
+                ... on PromptChatTemplate {
+                  messages {
+                    ... on TextPromptMessage {
+                      content
+                      role
+                    }
+                  }
+                }
+              }
+              tools {
+                __typename
+                definition
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const usePlaygroundPromptReference = (instanceId: number) => {
```

Comment:
I'd add a bit of docs here - it's a bit hard to grok as a first time reader of this code to understand that htis is constructing a query reference based on what's stored in the zustand store

Reply:
Agree. Will do that. In particular we need a reference instead of a lazy load so that we can conditionally perform the query so that would be good to document

Reply:
Removed in favor of an imperative fetch callback